File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        sum=0
        res=0
        dic={0:1}
        for i in nums:
            sum+=i
            diff=sum-k
            if diff in dic:
                res+=dic[diff]
            dic[sum]=dic.setdefault(sum,0)+1
        return res
        

File: Data Structures & Algorithms/remove-element/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums: number[], val: number): number {
        let k=0
        for(let i in nums){
            if(nums[i]!=val){
                nums[k]=nums[i]
                k++
            }
        
        }
        return k
    }
}